server/BudgetTracker.Data/Repositories/BudgetRepository.cs

```diff
@@ -145,7 +149,7 @@ public async Task LoadSubBudgets(Budget budget, bool recursive=false)
             budget.SubBudgets = subBudgets;
         }
 
-        private BudgetDurationModel GetBudgetDurationUpdated(BudgetDurationModel original, BudgetDurationBase newBudget)
+        private void SetNewBudgetDurationValues(BudgetDurationModel original, BudgetDurationBase newBudget)
```

Comment:
Why change this to void?

Reply:
To get rid of the side effect. Before, the function was editing in place. That worked for how I was using it before, but I wanted to re-use it here. Well, because of the side effect of it setting the property, it no longer worked. So I re-did it to make it just fetch the value. That way, I can do whatever I want with the value without worrying about any side effects that the function did in the process.

Reply:
Sounds good